### PR TITLE
Update aerospace extension

### PR DESCRIPTION
**Pull request by alexcamlo**
## Description

Adds a new command to change apps of the current workspace.

## Screencast

CleanShot 2024-10-11 at 20 05 16

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/aerospace/package-lock.json
Comment (AmmarCodes) [inline extensions/aerospace/package-lock.json:1]: Not sure why this file was included in the commit, I think it should be removed.

File: extensions/aerospace/CHANGELOG.md
Comment (AmmarCodes) [inline extensions/aerospace/CHANGELOG.md:3]:
```suggestion
## [New Feature] - 2024-10-11
```

**Review by AmmarCodes — APPROVED** [review]

File: extensions/aerospace/package-lock.json
Comment (thewinger) [reply extensions/aerospace/package-lock.json:1]: In Raycast extension guidelines: “ Please use npm for installing dependencies and include package-lock.json in your pull request.”

**Comment by AmmarCodes** [comment]
I understand. My point is about this PR not adding any dependencies yet
updating the lock file. We can leave this for the maintainer if you prefer.

**Comment by thewinger** [comment]
Honestly, I’m not knowledgeable enough. So whatever is the best thing to do.

Alejandro Cámara López
Interaction Designer
alejandrocamara.info

File: extensions/aerospace/package-lock.json
Comment (limonkufu) [reply extensions/aerospace/package-lock.json:1]: I should have included it myself,  thanks

**Review by limonkufu — APPROVED** [review]

**Comment by nikitabobko** [comment]
I don't have control over this PR, but

File: extensions/aerospace/assets/scripts/list-windows.sh
Comment (nikitabobko) [inline extensions/aerospace/assets/scripts/list-windows.sh:7]: I don't have control over this PR, and the final say is after people who support the raycast plugin. 

But as a maintainer of AeroSpace, the hacking around unstructured format makes me sad. Sure, you can use it for your own needs, but when you publish it for other people, I think it's better to clean it up. I'd wait for https://github.com/nikitabobko/AeroSpace/issues/577 before merging this PR

I want to take this opportunity to remind that the plugin currently parses `aerospace.toml` config itself (which is fragile), while the proper API was provided https://github.com/nikitabobko/AeroSpace/issues/215#issuecomment-2119209902 (and still waits to be used)

This PR increases the technical debt of the raycast plugin

**Comment by limonkufu** [comment]
Hey @nikitabobko,

I was afraid you would comment and remind me that I still been to change
that logic.

Family and other things happened so I couldn't do it. I am planning to do
it soon though hopefully in two weeks 😬

I am glad for you to give me a proper way and apologise again for my
tardiness

File: extensions/aerospace/assets/scripts/list-windows.sh
Comment (thewinger) [reply extensions/aerospace/assets/scripts/list-windows.sh:7]: List-windows.sh shouldn’t have been added. The command uses array-windows.sh but your comment still applies.
I don’t mind waiting if it will improve the code and technical debt. I started the PR before seeing the —json improvement.

**Comment by thewinger** [comment]
Closed in favor of updated version with Aerospace 0.15.0.

**Comment by joshgoebel** [comment]
So is this switchApps functionality coming soon then?  I was just working on adding it myself - would be great if it's already in the works!

**Comment by thewinger** [comment]
> So is this switchApps functionality coming soon then? I was just working on adding it myself - would be great if it's already in the works!

Yes, there is already a new PR with this functionality and aerospace latest —json improvement.

**Pull request closed without merging**
